File: single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& A) {
        int result = 0, count;
        for(int  i=0; i<32; i++)
        {
            count =0;
            for(int j =0 ; j< A.size(); j++)
            {
                if((A[j] >> i) & 1 == 1)
                {
                    count++;
                }
            }
            result+=(count%3)<<i;
        }
        return result;
    }
};

File: single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer, Integer> map= new HashMap<>();
        for(int i=0; i<nums.length; i++){
            if(map.containsKey(nums[i])) map.put(nums[i], map.get(nums[i])+1);
            else map.put(nums[i], 1);
        }
        for(Integer key : map.keySet())
        if(map.get(key)==1)
            return key;
        return -1;
    }
}